models/DataObject/AbstractObject.php

```diff
@@ -355,7 +355,7 @@ public static function getList($config = [])
     {
         $className = DataObject::class;
         // get classname
-        if (!in_array(static::class, [__CLASS__, Concrete::class], true)) {
+        if (!in_array(static::class, [__CLASS__, Concrete::class, Folder::class], true)) {
```

Comment:
@dvesh3 well this is a problem if a bundle using model class overrides (eg overriding Folder class extending original Pimcore Folder class will make this condition truthy).

Reply:
Would `self::OBJECT_TYPE_FOLDER` work?

Reply:
No...getType isn't static

should be something like this

```
        // get classname
        if (!in_array(static::class, [__CLASS__, Concrete::class, Folder::class], true)) {
            /** @var Concrete $tmpObject */
            $tmpObject = new static();
            if ($tmpObject instanceof Concrete) {
                $className = 'Pimcore\\Model\\DataObject\\' . ucfirst($tmpObject->getClassName());
            }
        }
```

Reply:
thanks @ssiergl I'll go with your suggested fix. Would you like to provide a PR?

Reply:
@dvesh3 will do this evening